qiskit/pulse/compiler/target.py

```diff
@@ -0,0 +1,306 @@
+
+    We make a native assumption that such hardware devices provide unique string identifier
+    for each component that it equips, and there are limited number of
+    such physical resources. We also assume a frame is tie to ports on the hardware,
+    namely, frames don't exist standalone and always form a mixed frame with a port in pair.
+
+    See :class:`.QiskitPulseTarget` for the reference implementation.
+    """
+
+    def get_frame_identifier(
+        self,
+        frame: model.Frame,
+    ) -> str:
+        """Return the unique string identifier of the frame.
+
+        Args:
+            frame: Qiskit pulse Frame object to inquire.
+
+        Returns:
+            A unique identifier of given frame.
+        """
+        raise NotExistingComponent(
+            f"This hardware doesn't proivde any resource implementing {frame}."
+        )
+
+    def get_port_identifier(
+        self,
+        port: model.LogicalElement,
+    ) -> str:
+        """Return the unique string identifier of the hardware port.
+
+        Args:
+            port: Qiskit pulse LogicalElement object to inquire.
+
+        Returns:
+            A unique identifier of given port.
+        """
+        raise NotExistingComponent(
+            f"This hardware doesn't proivde any resource implementing {port}."
+        )
+
+    def is_mixed_frame_available(
+        self,
+        mixed_frame: model.MixedFrame,
+    ) -> bool:
+        """Check if given mixed frame is implementable on the hardware.
+
+        Args:
+            mixed_frame: Qiskit pulse MixedFrame object to test.
+
+        Returns:
+            True if given mixed frame is implementable.
+        """
+        return False
+
+    def filter_mixed_frames(
+        self,
+        *,
+        frame: model.Frame | None = None,
+        port: model.LogicalElement | None = None,
+    ) -> list[model.MixedFrame]:
+        """Filter available mixed frames on the hardware.
+
+        Args:
+            frame: Qiskit pulse Frame object to include.
+            port: Qiskit pulse LogicalElement object to include.
+
+        Returns:
+            A list of Qiskit pulse MixedFrame objects that include given frame and port.
+        """
+        return []
+
+    def extra_frames(
+        self,
+        port: model.LogicalElement,
+    ) -> list[model.GenericFrame]:
+        """Get a list of string identifier of unused frames tied to the given port.
+
+        Args:
+            port: Qiskit pulse LogicalElement object to inquire.
+
+        Returns:
+            A list of unique identifier of unsed frames.
+        """
+        return []
+
+
+class QiskitPulseTarget(PulseTarget):
+    """Qiskit reference implementation of :class:`.PulseTarget`."""
+
+    def __init__(
+        self,
+        qubit_frames: dict[int, str] | None = None,
+        meas_frames: dict[int, str] | None = None,
+        qubit_ports: dict[int, str] | None = None,
```

Comment:
I would accept `meas_ports` as well to be more generic.

Reply:
Yes, that makes sense. PR is updated but in more general fashion.